### PR TITLE
Refactor/migrate to backend v2

### PR DESCRIPTION
**Pull request by swh00tw**
Migrate to v2 backend and update object structures to latest version

### CODE REVIEW

File: src/utils/getNolUrls.js [hunk 1]
```diff
@@ -6,14 +6,16 @@ const getNolUrl = (course) => {
   const lang = "CH";
   const base_url = "https://nol.ntu.edu.tw/nol/coursesearch/print_table.php?";
   const course_id = course.identifier.replace("E", "");
+  const dept_id = course.departments.length > 0 ? course.departments[0].id : "";
+  const class_id = course.class ? course.class : "";
```
Comment (swh00tw) [inline src/utils/getNolUrls.js:10]: Try to use optional chaining

```suggestion
  const dept_id =  course.departments?.[0]?.id ?? "";
  const class_id = course?.class ?? "";
```

File: src/components/CourseSearchInput.js [hunk 2]
```diff
@@ -143,12 +145,9 @@ function CourseSearchInputTextArea() {
             欄位
           </MenuButton>
           <MenuList>
-            <MenuOptionGroup
-              defaultValue={["course_name", "teacher"]}
-              type="checkbox"
-            >
+            <MenuOptionGroup defaultValue={["name", "teacher"]} type="checkbox">
```
Comment (jc-hiroto) [inline src/components/CourseSearchInput.js:148]: can you change all five fields' default values to enabled? thanks!

File: src/pages/api/user/removeFavoriteCourse.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { removeFavoriteCourse } from "queries/user";
+import { getAccessToken, withApiAuthRequired } from "@auth0/nextjs-auth0";
+import { assertNotNil } from "utils/assert";
+
+export default withApiAuthRequired(async function handler(req, res) {
+  try {
+    const { course_id } = req.body;
+    const { accessToken } = await getAccessToken(req, res);
+    if (!assertNotNil(course_id)) {
+      res.status(400).json({ error: "Missing user_id" });
```
Comment (jc-hiroto) [inline src/pages/api/user/removeFavoriteCourse.js:10]: you mean "Missing course_id" here?

File: src/pages/api/user/addFavoriteCourse.js [hunk 4]
```diff
@@ -4,17 +4,16 @@ import { assertNotNil } from "utils/assert";
 
 export default withApiAuthRequired(async function handler(req, res) {
   try {
-    const { new_favorite_list, user_id } = req.body;
+    const { course_id } = req.body;
     const { accessToken } = await getAccessToken(req, res);
-    if (!assertNotNil(user_id) || !assertNotNil(new_favorite_list)) {
+    if (!assertNotNil(course_id)) {
       res.status(400).json({ error: "Missing user_id" });
```
Comment (jc-hiroto) [inline src/pages/api/user/addFavoriteCourse.js:10]: you mean "Missing course_id" here?

**Review by jc-hiroto — CHANGES_REQUESTED** [review]

File: src/utils/getNolUrls.js (hunk 1 above)
Comment (jc-hiroto) [reply src/utils/getNolUrls.js]: Resolved, thanks!

**Review by jc-hiroto — APPROVED** [review]
Nice work Frank!

**Comment by swh00tw** [comment]
> Nice work Frank!

u too

**Pull request merged**